modules/context/repo.go

```diff
@@ -667,13 +667,37 @@ func RepoAssignment(ctx *Context) (cancel context.CancelFunc) {
 	}
 	ctx.Data["Tags"] = tags
 
-	brs, _, err := ctx.Repo.GitRepo.GetBranchNames(0, 0)
+	branchOpts := git_model.FindBranchOptions{
+		RepoID:          ctx.Repo.Repository.ID,
+		IsDeletedBranch: util.OptionalBoolFalse,
+		ListOptions: db.ListOptions{
+			ListAll: true,
+		},
+	}
+	branchesTotal, err := git_model.CountBranches(ctx, branchOpts)
+	if err != nil {
+		ctx.ServerError("CountBranches", err)
+		return
+	}
+
+	// non empty repo should have at least 1 branch, so this repository's branches haven't been synced yet
+	if branchesTotal == 0 { // fallback to do a sync immediately
+		if err := repo_module.SyncRepoBranches(ctx, ctx.Repo.Repository.ID, 0); err != nil {
```

Comment:
IIRC, there is also a RepoAssignment for API handlers, I think that RepoAssignment also needs this logic?

Reply:
But not all endpoints of API will visit branches.

Reply:
I added the same sync logic for API list branches and delete branch.